### ISSUE TITLE
Lineage service

### ISSUE THREAD

**Issue by mcdoker18** [odahu/odahu-flow#179]
TODO: fill description

**Comment by karbyshevds** [comment]
We need to provide separate API that will allow users to pass metadata from different ODAHU workflow steps to it using unique identifier. The main goal is to be able to track all workflow activity by ID on all steps.

**Comment by alinaignatiuk** [comment]
Non-trivial logging structured for particular pipeline which allows tracking steps status across the ML pipeline (prediction, production pipelines). 
The main goal is debugging to understand what goes wrong
The next is to regulation of the processes inside (what is going on and whether processes are compliant)
3r party solutions primary (MLflow logging, sagemaker) goal is to track the ML specific steps status
kubeflow pipeline
need research for https://github.com/google/ml-metadata, however we should into account ODAHU does not have own pipeline service
orchestration service?
LP
